src/Cardano/Wallet/BlockSyncer.hs

```diff
@@ -0,0 +1,65 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+
+-- |
+-- Copyright: © 2018-2019 IOHK
+-- License: MIT
+--
+-- This module contains the ticking function that is responsible for invoking block acquisition functionality and executing it in periodic fashion.
+--
+-- Known limitations: ticking function makes sure action is not executed on already consumed block, but does not check and handle block gaps (aka catching up)
+
+module Cardano.Wallet.BlockSyncer
+  (
+    BlockHeadersConsumed(..)
+  , tickingFunction
+  ) where
+
+
+import Prelude
+
+import Cardano.Wallet.Primitive
+    ( Block (..), BlockHeader )
+import Control.Concurrent
+    ( threadDelay )
+import Data.Time.Units
+    ( Second, toMicroseconds )
+
+import qualified Data.List as L
+
+
+newtype BlockHeadersConsumed = BlockHeadersConsumed [BlockHeader] deriving (Show, Eq)
+
+storingLimit :: Int
+storingLimit = 2140
```

Comment:
Did you mean `2160` :thinking: ?

Reply:
yeah, when we later tackle rollback security parameter comes here (I thought about it but made blunder here). I wanted to just add some upper bound just in order to not surpass memory limit.